### PR TITLE
Add pretraining loss computation for TF Bert pretraining

### PR DESCRIPTION
**Pull request by jplu**
# What does this PR do?

This PR adds the loss computation for the pretraining TF BERT model. The loss computation test is also more robust on variable call signature lengths.

### CODE REVIEW

File: src/transformers/modeling_tf_bert.py [hunk 1]
```diff
@@ -843,17 +890,44 @@ def call(self, inputs, **kwargs):
...
-        outputs = self.bert(inputs, **kwargs)
+
+        if isinstance(inputs, (tuple, list)):
+            labels = inputs[9] if len(inputs) > 9 else labels
+            next_sentence_label = inputs[10] if len(inputs) > 10 else next_sentence_label
+            if len(inputs) > 9:
+                inputs = inputs[:9]
+        elif isinstance(inputs, (dict, BatchEncoding)):
+            labels = inputs.pop("labels", labels)
+            next_sentence_label = inputs.pop("next_sentence_label", next_sentence_label)
```
Comment (LysandreJik) [inline src/transformers/modeling_tf_bert.py:902]: Just realized that this seems to be an issue in all TF models: if `return_dict` is defined in the `inputs` (they're either a tuple, a list or a dict), the value of `return_dict` won't be used here?

File: src/transformers/modeling_tf_t5.py [hunk 2]
```diff
@@ -1268,13 +1267,6 @@ def call(
         else:
             input_ids = inputs
 
-            if "past_key_value_states" in kwargs:
-                warnings.warn(
-                    "The `past_key_value_states` argument is deprecated and will be removed in a future version, use `past_key_values` instead.",
-                    FutureWarning,
-                )
-                past_key_values = kwargs.pop("past_key_value_states")
-
```
Comment (LysandreJik) [inline src/transformers/modeling_tf_t5.py:1277]: No need to remove this until we change major version.

File: src/transformers/modeling_tf_bert.py (hunk 1 above)
Comment (jplu) [reply src/transformers/modeling_tf_bert.py:902]: Yep! But will be solved with the new input parsing. A fix will be in a PR that will arrive soon.

File: src/transformers/modeling_tf_t5.py (hunk 2 above)
Comment (jplu) [reply src/transformers/modeling_tf_t5.py:1277]: The `**kwargs` parameter has to be removed anyway (it brings few issues) so it should be a good move to remove it for the major version.

File: tests/test_modeling_tf_common.py [hunk 3]
```diff
@@ -98,6 +99,14 @@ def _prepare_for_class(self, inputs_dict, model_class, return_labels=False) -> d
...
             elif model_class in TF_MODEL_FOR_NEXT_SENTENCE_PREDICTION_MAPPING.values():
                 inputs_dict["next_sentence_label"] = tf.zeros(self.model_tester.batch_size, dtype=tf.int32)
+            elif model_class in TF_MODEL_FOR_PRETRAINING_MAPPING.values():
+                # Only BERT needs the next sentence label for pre-training
+                if model_class.base_model_prefix in ["bert"]:
+                    inputs_dict["next_sentence_label"] = tf.zeros(self.model_tester.batch_size, dtype=tf.int32)
+
+                inputs_dict["labels"] = tf.zeros(
+                    (self.model_tester.batch_size, self.model_tester.seq_length), dtype=tf.int32
+                )
```
Comment (LysandreJik) [inline tests/test_modeling_tf_common.py:109]: If it's BERT specific, I'd prefer to override this in `TFBertModelTest` instead of putting model-specific stuff in a model-agnostic file

**Review by LysandreJik — APPROVED** [review]
LGTM other than one nitpick

File: tests/test_modeling_tf_common.py (hunk 3 above)
Comment (jplu) [reply tests/test_modeling_tf_common.py:109]: How would you do that? As this is only for BERT + PRetraining.

Comment (jplu) [reply tests/test_modeling_tf_common.py]: @LysandreJik What do you think of this approach to solve this?

Comment (LysandreJik) [reply tests/test_modeling_tf_common.py]: No, this isn't what I had in mind. I think overriding this in `TFBertModelTest` would be better. Similarly to how the `test_saved_model_with_hidden_states_output` is overriden in `TFLxmertModelTest`: redefining the test in the tester when it diverges to the common test.

We should always extrapolate that a model-specific change will lead to other model-specific changes down the road, and this would become unmaintainable. Overriding in the model-specific tester makes much more sense from a maintainability point of view.

Comment (jplu) [reply tests/test_modeling_tf_common.py]: Ok got it! Sorry I wasn't really how to overwrite a test.

Comment (jplu) [reply tests/test_modeling_tf_common.py]: Just made an update. Does-it looks better?

**Review by LysandreJik — APPROVED** [review]
Perfect, thanks for iterating!
